src/Common/src/System/Security/Cryptography/ECDiffieHellmanDerivation.cs

```diff
@@ -0,0 +1,243 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace System.Security.Cryptography
+{
+    internal static class ECDiffieHellmanDerivation
+    {
+        /// <summary>
+        /// Derive the raw ECDH value into <paramref name="hasher"/>, if present, otherwise returning the value.
+        /// </summary>
+        internal delegate byte[] DeriveSecretAgreement(ECDiffieHellmanPublicKey otherPartyPublicKey, IncrementalHash hasher);
```

Comment:
Any reason to use a custom delegate over `Func<EECDiffieHellmanPublicKey, IncrementalHash, byte[]>`?

Reply:
Partially as a place to leave the comment of null is expected when hasher isn't.  (And maybe the act of writing the commend helped me fix whatever problem I had been having at the time).

If we're anti-non-public-delegates I can probably switch (back) to Func now.